### ISSUE TITLE
[BUG] Unable to reset password from reset password link

### ISSUE THREAD

**Issue by filipedelimabrito** [RocketChat/Rocket.Chat#11859]
I tried several times reseting my password after getting its link on my e-mail address (by sending my e-mail from the Reset Password screen on the Android App and from the web). Even successfully receiving the reset password link I always get this error message:

<img width="595" alt="screen shot 2018-08-24 at 09 29 15" src="https://user-images.githubusercontent.com/11776825/44585000-2ea53980-a781-11e8-8c24-3f2849dda6db.png">

- Version of Rocket.Chat Server: https://open.rocket.chat
- Browser: Safari

**Comment by engelgabriel** [comment]
@sampaiodiego do we need a patch to fix this?

**Comment by sampaiodiego** [comment]
if we happen to have a fix, yes. I guess it's a frontend issue since it only happens on safari

**Comment by Adam212** [comment]
Hi,

It happens under any browser (tried FF and Chromium)
And because of bug #11884 it's now also impossible to change password by admin.

**Comment by engelgabriel** [comment]
@ggazzo can you please look at this. Looks like an extremely high priority bug.

**Comment by rodrigok** [comment]
As I tested here, if you try to reset the password if you are logged in then you receive that error, otherwise it works.

The code involved was not changed for a long time.

We may check for the non existence of `FlowRouter.getParam('token')` here https://github.com/RocketChat/Rocket.Chat/blob/cecf105601b74373e283e491a74c107a0c3a0008/packages/rocketchat-ui-login/client/reset-password/resetPassword.js#L32 in addition of the user been logged, so if the token was passed it goes through the reset flow even when logged.

**Comment by IITNTII** [comment]
Sorry, I just test it in https://open.rocket.chat it still happens.
I got this error when I test in my local
```
Exception while invoking method 'resetPasswordWithTOTP' Error: Can't call setUserId in a method after calling unblock
```

**Comment by Hudell** [comment]
That seems to be a problem with the two factor authentication, do you mind opening a new issue?

**Comment by IITNTII** [comment]
New issue created: https://github.com/RocketChat/Rocket.Chat/issues/12820
